### PR TITLE
Add jitter to connect reuse timeout

### PR DESCRIPTION
**Pull request by adilhafeez**
add 1.3x jitter to socket expiry time

### CODE REVIEW

**Comment by jsnoop36** [comment]
👍

File: server.go [hunk 1]
```diff
@@ -185,7 +186,9 @@ func (server *mongoServer) Connect(timeout time.Duration) (*mongoSocket, error)
 	stats.conn(+1, master)
 	var socketExpiryTime *time.Time
 	if server.maxSocketReuseTime != 0 {
-		expiryTime := time.Now().Add(server.maxSocketReuseTime * time.Second)
+		duration := server.maxSocketReuseTime * time.Second
+		durationWithJitter := time.Duration(float64(duration) * (1 + rand.Float64()*.3))
```
Comment (rodaine) [inline server.go]: can we convert the jitter value to a constant to avoid the magic number?

**Review by rodaine — APPROVED** [review]
Unrelated to your change but it's problematic that maxSocketReuseTime is stored as a `time.Duration` and is later multiplied by `time.Second` here to get the actual duration. That computation should be done when the `server` is created

**Comment by adilhafeez** [comment]
@rodaine that is not nice, have fixed it - thanks for pointing it out

**Pull request merged**